AK/JsonParser.cpp

```diff
@@ -48,32 +48,48 @@ String JsonParser::consume_and_unescape_string()
         if (next_is('\0')) {
             return {};
         }
+        
+        if (next_is('n')) {
             final_sb.append('\n');
-            break;
-        case 'r':
+            continue;
+        }
+        if (next_is('r')) {
             final_sb.append('\r');
-            break;
-        case 't':
+            continue;
+        }
+        if (next_is('t')) {
             final_sb.append('\t');
-            break;
-        case 'b':
+            continue;
+        }
+        if (next_is('b')) {
             final_sb.append('\b');
-            break;
-        case 'f':
+            continue;
+        }
+        if (next_is('f')) {
             final_sb.append('\f');
-            break;
-        case 'u': {
+            continue;
+        }
+        if (next_is('u')) {
+            if (tell_remaining() < 4) {
+                return {};
+            }
+
             auto code_point = AK::StringUtils::convert_to_uint_from_hex(consume(4));
-            if (code_point.has_value())
+            if (code_point.has_value()) {
                 final_sb.append_code_point(code_point.value());
-            else
-                final_sb.append('?');
-        } break;
-        default:
-            final_sb.append(escaped_ch);
+                continue;
+            } else {
+                return {};
+            }
+        }
+
+        if (!tell_remaining()) {
```

Comment:
And here. Also, please consider explicitly stating `tell_remaining() == 0`, since we're not working with a boolean or error return code here. It just makes it more clear to the reader what you are doing.